include/swift/AST/DiagnosticsSema.def

```diff
@@ -825,6 +825,10 @@ ERROR(argument_out_of_order_unnamed_unnamed,none,
       "unnamed argument #%0 must precede unnamed argument #%1",
       (unsigned, unsigned))
 
+ERROR(member_shadows_global_function,none,
+      "member '%0.%2' shadows global function '%1.%2'",
```

Comment:
Thanks, I think this will be much more helpful to users. Thinking about it more, it's still not quite enough information to allow someone to understand *why* it's an error. (If %0 shadows %1, why doesn't the compiler just treat it as a call to %0?)

An idea to make this even better: change `member` → `static member` (or perhaps `class member` as appropriate, or just `type member`), and add `, but cannot be called with implicit 'self.'` similar to what you had originally.

It might also be nice to include a fix-it that inserts `MyType.` or `type(of: self).` (assuming those would work).